lib/dependency-manager-adapters/npm.js

```diff
@@ -54,10 +73,19 @@ module.exports = CoreObject.extend({
   },
   cleanup: function() {
     var adapter = this;
-    return adapter._restoreOriginalDependencies().then(function() {
+    var backupYarnLockPresent = fs.existsSync(path.join(this.cwd, this.yarnLockBackupFileName));
+    return this._restoreOriginalDependencies(backupYarnLockPresent).then(function() {
       debug('Remove backup package.json and node_modules');
-      return RSVP.all([rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
-        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))]);
+
+      var cleanupTasks = [rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
+        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))];
+
+      if (backupYarnLockPresent) {
+        cleanupTasks.push(rimraf(path.join(adapter.cwd, adapter.yarnLockBackupFileName)));
+        adapter.noLock = false;
```

Comment:
By the time this `noLock` flag is set to false, the reinstall of original dependencies will have been completed.

For the project that has a yarn.lock, we want to run scearios WITHOUT a lockfile, but upon cleanup use that lockfile to restore them to their normal, locked, state.